src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -271,27 +273,31 @@ impl FlowServiceImpl {
                 trigger_time: start_time,
             }),
             FlowTriggerContext::Unconditional,
+            ConfigSnapshot::default(),
         )
         .await
     }
 
-    #[tracing::instrument(level = "trace", skip_all, fields(?flow_key, %flow_id))]
+    #[tracing::instrument(level = "trace", skip_all, fields(?flow))]
     async fn enqueue_dependent_flows(
         &self,
         input_success_time: DateTime<Utc>,
-        flow_key: &FlowKey,
-        flow_id: FlowID,
+        flow: &Flow,
         flow_result: &FlowResult,
     ) -> Result<(), InternalError> {
-        if let FlowKey::Dataset(fk_dataset) = flow_key {
+        if let FlowKey::Dataset(fk_dataset) = &flow.flow_key {
             // Extract a list of downstream 1 level datasets
+            let flow_manual_trigger_initiator = flow
+                .try_get_manual_trigger()
+                .map(|trigger| trigger.initiator_account_id);
+
```

Comment:
Why only manual trigger? This isn't a general rule for all possible flow types

Reply:
Changed